controllers/storagecluster/cephcluster.go

```diff
@@ -461,9 +461,13 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, nodeCount int, s
 				rookCephv1.KeyOSD: systemNodeCritical,
 			},
 			Resources: newCephDaemonResources(sc),
-			ContinueUpgradeAfterChecksEvenIfNotHealthy: true,
+			// if resource profile change is in progress, then set this flag to false
+			ContinueUpgradeAfterChecksEvenIfNotHealthy: sc.Spec.ResourceProfile == sc.Status.ResourceProfile,
```

Comment:
The `sc.Status.ResourceProfile` is only updated after the successful reconcile of the new resource profile, right? So this is comparing the new profile with the previous profile (if it changed)?

Reply:
Will this cause all upgraded clusters (the first time upgrading to 4.15) to set this to `false`? Let's keep this as `true` if we're applying the default profile and if the applied profile is blank.

Reply:
Yeah, the thought was that if the values on the status & the spec are not the same that means we are in the process of applying the resource profile that's in the spec.

Reply:
No, when someone upgrades to 4.15, They don't have the resource profile either in their spec or status. So two blank values are equal, and the flag would be true only.